app/code/Magento/Widget/Block/Adminhtml/Widget/Options.php

```diff
@@ -166,7 +166,9 @@ protected function _addField($parameter)
 
         if (is_array($data['value'])) {
             foreach ($data['value'] as &$value) {
-                $value = html_entity_decode($value);
+                if (!is_array($value)) {
```

Comment:
It would be better to check the value on string type: if(is_string($value)) because this check will fail if $value will be an object

Reply:
@sidolov makes sense. Thank you.